### DISCUSSION TITLE
Request to Improve Clarity of Random Word Alias After Recent Commit

### DISCUSSION THREAD

**Discussion by sehgal-aamulya** [simple-login/app#1660] [Feature Requests]
Hi,

After the merge of pull request: #1625 , the format of random word alias has been changed to:
_<random_word1><random_word2><random_numbers>_@alias.tld.

I have no issue with the random numbers in the alias but would request the addition of a word separator like **_** (as it existed for _SimpleLogin_ before the pull request) or **.** (as it exists for competitors like _AnonAddy_ and _FastMail Masked Address_) for more clarity.

The requested format for the random word alias:
_<random_word1>_**_**_<random_word2><random_number>_@alias.tld
or
_<random_word1>_**.**_<random_word2><random_number>_@alias.tld

There has been a discussion on this in #1645 and https://www.reddit.com/r/Simplelogin/comments/11u5f9z/random_alias_generator/.

Hope this is can be considered.

**Comment by nguyenkims** [comment]
@sehgal-aamulya hey currently the format is `<random_word1><random_number>@alias.tld` so only 1 random word is used.

**Reply by sehgal-aamulya** [reply]
Hi @nguyenkims ,

Thanks for the response.

Are you sure the format is _<random_word1><random_number>@alias.tld_? 

As per https://github.com/simple-login/app/blob/1f9d7843828ed2ae0d218652fef2af08b0cb23f8/app/models.py#L1276 and https://github.com/simple-login/app/blob/1f9d7843828ed2ae0d218652fef2af08b0cb23f8/app/utils.py#L29 it seems to be _<random_word1><random_word2><random_numbers>@alias.tld_

<img width="1791" alt="Screenshot 2023-03-28 at 2 18 26 PM" src="https://user-images.githubusercontent.com/35224269/228370297-4140e285-2f36-4a56-bb79-89fc368c9c24.png">

Also, please see the attached screenshot, it doesn't seem like "bountifulcornflake" and "gossipcrewless" are one words but two words like "bountiful" and "cornflake"; "gossip" and "crewless".

Can you please check and confirm?

Thank you

**Reply by sehgal-aamulya** [reply]
Hi @nguyenkims ,

just to confirm this for aliases created using the feature "**Random Aliases By Random Words**" and not creating a _custom new alias_ which has a _random word number suffix_.

Thank you

**Reply by sehgal-aamulya** [reply]
Hi @nguyenkims ,

Please check the links in the description again if possible, pull request #1625 , discussion https://github.com/simple-login/app/discussions/1645#discussioncomment-5408876 and reddit discussion https://www.reddit.com/r/Simplelogin/comments/11u5f9z/random_alias_generator/.  They might help by providing more clarity.

The pull request #1625 changed the format for **generated random emails** that use **random words**. It seems like maybe this was not the intention and that change maybe should have been restricted to only the _random word number suffix_ for _custom/new alias_.

It would be great if we could go back to the original format of:
_<random_word1>_**_**_<random_word2>_@alias.tld
or if numbers are needed then:
_<random_word1>_**_**_<random_word2><random_number>_@alias.tld

I hope this helps and a decision can be taken to correct/improve the format to be more _readable and clear_.

_PS: Excuse me for repeated posts on this_

Thank you

**Reply by nguyenkims** [reply]
Oh I understand now, thanks for the clarification. It should be done in https://github.com/simple-login/app/pull/1673

**Comment by ghost** [comment]
@nguyenkims can you also provide ability to generate totally random aliases like `589mjcsbf23h@domain.com`. This feature is available from your competitors (e.g. AnonAddy. Duck, Firefox Relay). This feature will be useful to completely confuse the tracking party and reduce the "consumption" of beautiful addresses which, as you said, cannot be reused

**Reply by sehgal-aamulya** [reply]
I believe you can generate random UUID based aliases. Does that work for you?
